pkg/controller/replicaset/replica_set.go

```diff
@@ -515,8 +551,13 @@ func (rsc *ReplicaSetController) manageReplicas(filteredPods []*v1.Pod, rs *apps
 		}
 		klog.V(2).Infof("Too many replicas for %v %s/%s, need %d, deleting %d", rsc.Kind, rs.Namespace, rs.Name, *(rs.Spec.Replicas), diff)
 
+		relatedPods, err := rsc.getIndirectlyRelatedPods(rs)
+		if err != nil {
+			return err
```

Comment:
A return here probably isn't appropriate.  I think you should just log the error here at utilruntime.HandleError level.  If you can't fall back to related pods you are strictly no worse than current code.  Also, add a test case validating that for getPodsToDelete (relatedPods empty) just far sanity if you haven't.

Reply:
Fixed.  `utilruntime.HandleError` does nothing if given a nil value, so I replaced the whole `if` statement with `utilruntime.HandleError(err)`.